integrationtests/src/test/java/io/kroxylicious/proxy/multitenant/MultiTenantIT.java

```diff
@@ -166,21 +165,56 @@ public void publishOne(KafkaCluster cluster) throws Exception {
-            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, MY_KEY, MY_VALUE, false);
+            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, groupName, MY_KEY, MY_VALUE, false);
         }
     }
 
     @Test
     public void consumeOneAndOffsetCommit(KafkaCluster cluster) throws Exception {
         String config = getConfig(PROXY_ADDRESS, cluster);
         try (var proxy = startProxy(config)) {
+            var groupName = testInfo.getDisplayName();
             createTopics(TENANT1_PROXY_ADDRESS, List.of(NEW_TOPIC_1));
             produceAndVerify(TENANT1_PROXY_ADDRESS,
                     Stream.of(new ProducerRecord<>(TOPIC_1, MY_KEY, MY_VALUE + "1"), new ProducerRecord<>(TOPIC_1, MY_KEY, MY_VALUE + "2")));
-            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, MY_KEY, MY_VALUE + "1", true);
-            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, MY_KEY, MY_VALUE + "2", true);
+            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, groupName, MY_KEY, MY_VALUE + "1", true);
+            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, groupName, MY_KEY, MY_VALUE + "2", true);
+        }
+    }
+
+    @Test
+    public void alterOffsetCommit(KafkaCluster cluster) throws Exception {
+        String config = getConfig(PROXY_ADDRESS, cluster);
+        try (var proxy = startProxy(config); var admin = Admin.create(commonConfig(TENANT1_PROXY_ADDRESS, Map.of()))) {
+            var groupName = testInfo.getDisplayName();
+            createTopics(TENANT1_PROXY_ADDRESS, List.of(NEW_TOPIC_1));
+            produceAndVerify(TENANT1_PROXY_ADDRESS,
+                    Stream.of(new ProducerRecord<>(TOPIC_1, MY_KEY, MY_VALUE + "1"), new ProducerRecord<>(TOPIC_1, MY_KEY, MY_VALUE + "2"),
+                            new ProducerRecord<>(TOPIC_1, MY_KEY, MY_VALUE + "youshouldntconsumeme")));
+            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, groupName, MY_KEY, MY_VALUE + "1", true);
+            var rememberedOffsets = admin.listConsumerGroupOffsets(groupName).partitionsToOffsetAndMetadata().get();
+            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, groupName, MY_KEY, MY_VALUE + "2", true);
+
+            admin.alterConsumerGroupOffsets(groupName, rememberedOffsets).all().get();
+            consumeAndVerify(TENANT1_PROXY_ADDRESS, TOPIC_1, groupName, MY_KEY, MY_VALUE + "2", true);
+        }
+    }
+
+    @Test
+    public void deleteConsumerGroupOffsets(KafkaCluster cluster) throws Exception {
+        String config = getConfig(PROXY_ADDRESS, cluster);
+        try (var proxy = startProxy(config); var admin = Admin.create(commonConfig(TENANT1_PROXY_ADDRESS, Map.of()))) {
+            var groupName = testInfo.getDisplayName();
+            createTopics(TENANT1_PROXY_ADDRESS, List.of(NEW_TOPIC_1));
+            produceAndVerify(TENANT1_PROXY_ADDRESS,
+                    Stream.of(new ProducerRecord<>(TOPIC_1, MY_KEY, MY_VALUE + "1"), new ProducerRecord<>(TOPIC_1, MY_KEY, MY_VALUE + "2")));
```

Comment:
Why do we produce to records to the topic but only verify that we read one?

Reply:
So to partly answer my own question we configure the consumer with `MAX_POLL_RECORDS_CONFIG` and set that to the number of expected records. Which explains why we only get a single result back to verify. However that doesn't really answer the more philosophical aspect of the question: What is the point of producing two records but then deliberately only consuming one of them?

Reply:
Hello! In earlier versions of this method, the deletion of our offset wasn't working as expected, so rather than consuming the record that was expected (1), it was just hanging until the test timed out. This took a good bit of debugging and turning up log levels to figure out what was even causing the hang. I think @k-wall put it best when he said that consuming an unexpected value (so 2, instead of 1) would be easier for a developer to debug in the future, than just a hanging test :)

Reply:
Yes, that's right @racheljpg.   We wanted a clear assertion failure tell us that this isn't the message we were expecting rather than a hang/timeout.

Reply:
There's another usage where the value is `youshouldntconsumeme`. Maybe if both usages called a method like:

```
ConsumerRecord backstop(String topic, String key) {
    return new ConsumerRecord<>(topic, key, "unexpected - should never be consumed")
}
```

It would show they're being used the same way and give a good place for a comment (though it's hard to name, I'm sure backstop isn't a very universal term)

Reply:
I like @robobario 's idea of pulling out a method or a constant to give the extra entry some descriptive power.

Reply:
Agree with @robobario, but I think this is related to the `alterOffsetCommit` test case as well

Reply:
Thanks, everyone! I've implemented a method that produces a message with a value like Rob suggested above, so if a test fails by consuming that one we know where to start